llvm/lib/DebugInfo/Symbolize/Symbolize.cpp

```diff
@@ -604,13 +604,13 @@ LLVMSymbolizer::createModuleInfo(const ObjectFile *Obj,
 }
 
 Expected<SymbolizableModule *>
-LLVMSymbolizer::getOrCreateModuleInfo(const std::string &ModuleName) {
-  std::string BinaryName = ModuleName;
-  std::string ArchName = Opts.DefaultArch;
+LLVMSymbolizer::getOrCreateModuleInfo(StringRef ModuleName) {
+  StringRef BinaryName = ModuleName;
+  StringRef ArchName = Opts.DefaultArch;
```

Comment:
Given that you have to convert them to `std::string` further down, and the syntax for that is a little uglier, I think it would just make sense to use the `std::string` type here like it was before.

Reply:
In my use case I have literally tens of millions of addresses to symbolize over ~50-100 different binaries (you can think of it as a hand-made analogue of `perf record -a` with subsequent `perf script`).

Not that it matters too much, but creating a `std::string BinaryName` from `ModuleName` (allocation + deallocation + memcpy) for every address is wasteful, given that the actual `std::string` is only needed ~50-100 times (the amount of binaries in question) to insert into maps.

Reply:
I'm curious to see whether the change has any significant performance impact. Do you have a way of measuring the time taken with and without this specific bit of the change, with a release build of the symbolizer?